src/Test/Html/Selector.elm

```diff
@@ -309,6 +354,54 @@ checked =
 -- HELPERS
 
 
+findAttributeInFacts : Bool -> String -> ElmHtml.InternalTypes.Facts a -> Selector
+findAttributeInFacts caseInsensitive name facts =
+    let
+        lookup =
+            case caseInsensitive of
+                True ->
+                    dictGetCaseInsensitive
+
+                False ->
+                    Dict.get
+    in
+    facts.stringAttributes
+        |> lookup name
+        |> Maybe.map (namedAttr name)
+        |> orElseLazy
+            (\_ ->
+                facts.boolAttributes
+                    |> lookup name
+                    |> Maybe.map (namedBoolAttr name)
+            )
+        |> Maybe.withDefault Invalid
+
+
+findInList : (a -> Bool) -> List a -> Maybe a
+findInList predicate list =
+    case list of
+        [] ->
+            Nothing
+
+        first :: rest ->
+            if predicate first then
+                Just first
+            else
+                findInList predicate rest
+
+
+dictGetCaseInsensitive : String -> Dict String a -> Maybe a
+dictGetCaseInsensitive key dict =
+    let
+        keyToLower =
+            String.toLower key
+    in
+    dict
+        |> Dict.toList
+        |> findInList (\( k, v ) -> String.toLower k == keyToLower)
```

Comment:
Yeah, the best way to see this library is way to test the correctness of the virtual-dom implementation. It is not intended to perfectly match a real DOM, since integration tests and QA is the best way to achieve that